pkg/operator/edgefs/isgw/isgw.go

```diff
@@ -139,7 +139,7 @@ func (c *ISGWController) makeISGWService(name, svcname, namespace string, isgwSp
 		},
 		Spec: v1.ServiceSpec{
 			Selector: labels,
-			Type:     v1.ServiceTypeNodePort,
+			Type:     v1.ServiceType(isgwSpec.ServiceType),
```

Comment:
There is already code to handle the error in the event the operator tries to create a bad service, although it is a bit lacking (for example, if a service fails to start, the pod will be left dangling).  
```
2019-07-25 13:21:02.926979 E | edgefs-op-s3: failed to create s3 s3-cola. failed to create s3 service. Service "rook-edgefs-s3-s3-cola" is invalid: spec.type: Unsupported value: "NidePort": supported values: "ClusterIP", "ExternalName", "LoadBalancer", "NodePort"
```

I had assumed that this would be sufficient, since it is the exact error a user would receive if they tried to create an off-spec Service in K8s.  I can certainly add checks to validateService() if you think that would be a better way of handling the input.

Reply:
that looks very similar to something I was doing before.  I'll bring it back.

Reply:
Added ServiceType check to the validateService() function